lib/events/index.js

```diff
@@ -13,23 +30,75 @@ function tophatEvents( document, tophatElement, serviceElement, config ) {
 }
 
 function attachTophatEvents( document, tophatElement ) {
-    utils.attachDomEvent( tophatElement, 'click', utils.proxy( tophatElement, clickhandler ) );
-    utils.attachDomEvent( document, 'click', utils.proxy( tophatElement, cancelhandler ) );
-    utils.attachDomEvent( tophatElement, 'click', utils.proxy( tophatElement, trackingHandler ) );
-    utils.attachDomEvent( tophatElement, 'submit', utils.proxy( tophatElement, searchHandler ) );
+    domUtils.attachDomEvent( tophatElement, 'click', domUtils.proxy( tophatElement, clickhandler ) );
+    domUtils.attachDomEvent( document, 'click', domUtils.proxy( tophatElement, cancelhandler ) );
+    domUtils.attachDomEvent( tophatElement, 'click', domUtils.proxy( tophatElement, trackingHandler ) );
+    domUtils.attachDomEvent( tophatElement, 'submit', domUtils.proxy( tophatElement, searchHandler ) );
+    domUtils.attachDomEvent( tophatElement, 'keydown', domUtils.proxy( tophatElement, keyHandler ) );
 
     // Throttle window resize event to 15fps
     var fps = 15,
         wait = Math.round(1000 / 15);
-    utils.attachDomEvent( window, 'resize', utils.throttle(utils.proxy( tophatElement, resizeHandler), wait) );
+    domUtils.attachDomEvent( window, 'resize', domUtils.throttle(domUtils.proxy( tophatElement, resizeHandler), wait) );
 }
 
 
-
-
 // handlers
 //
 
+// Handle key press to delegate action
+function keyHandler(e) {
+
+	// Get keycode, decide what to do based on the key
+	var target = e.target || e.srcElement,
+		keyCode = e.which || e.keyCode;
+
+	//Gets menuIds controlled by the element controlling the parentmenu
+	var parentMenuEl = domUtils.closestWithClass(target, "menu");
+	var menuIdsWithinScope = containerIds(parentMenuEl, domUtils);
+
+	if(keyCode === KeysDict.escape) {
+		this.state.unfocus(); // Close all menus
+		if(parentMenuEl) {
+			// Return focus to the controlling link for a sub menu
+			var controllerLinkId = parentMenuEl.getAttribute("aria-labelledby");
+			document.getElementById(controllerLinkId).focus();
+		}
+		return;
+	}
+	else if(keyCode === KeysDict.enter || keyCode === KeysDict.space) {
+		// Pressing space/enter 'activates' a link/button so use the click handler we already have
+		clickhandler.call(this, e);
+		return;
+	}
+
+	if(!menuIdsWithinScope) {
+		return;
+	}
+
+	switch(keyCode) {
+		// e.preventDefault();
+		case KeysDict.left:
+		case KeysDict.up:
+			domUtils.getPreviousLink(menuIdsWithinScope, target).focus();
+			break;
+		case KeysDict.right:
+		case KeysDict.down:
+			domUtils.getNextLink(menuIdsWithinScope, target).focus();
+			break;
+		case KeysDict.home:
+			domUtils.getLinks(menuIdsWithinScope)[0].focus();
+			break;
+		case KeysDict.end:
+			var links = domUtils.getLinks(menuIdsWithinScope);
+			links[links.length - 1].focus();
+			break;
+		default:
+			return;
+	}
+	e.preventDefault();
```

Comment:
Does e.preventDefault work in ie8? There's a `utils.enhanceEvent` function that seems to do some cross browser prevent default stuff which I wonder if we should be using.